### DISCUSSION TITLE
How to test a component rendered in a Controller?

### DISCUSSION THREAD

**Discussion by alan-nakagiri** [react-hook-form/react-hook-form#6174] [Q&A] [answered]
Greetings! I am trying to figure out how to write a test for a component that I have rendered in a `<Controller/`

I am using the Controller to render a `react-select` dropdown component like so: 

```
        <Controller
          name={name}
          control={control}
          rules={null}
          render={({ field }) => (
            <Select
              {...field}
              name={name}
              options={selectOptions}
              className={css(styles.dropdown, inputBorderError, dateStyles)}
              onChange={(e) => handleOnChange(field, e)}
              onBlur={(e) => handleOnBlur(field, e)}
              onFocus={(e) => handleOnFocus(e)}
              styles={customStyles}
              value={{ label: controlValue, value: controlValue }}
              isSearchable={false}
            />
          )}
        />
```

When I attempt to render the component in my jest test (enzyme or the react testing library) I get an error. I'm sure I am doing something wrong. 

Does anyone have tips on how to successfully rendered a component like this in a test where you could test the onChange/Blur/Focus functionality?

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
have a look at those: https://github.com/react-hook-form/react-hook-form/blob/master/src/__tests__/controller.test.tsx

**Reply by alan-nakagiri** [reply]
Got it thanks! I see that I have to render a Form that uses my component instead of just rendering the component. 

Thanks!

**Reply by DaviAraki** [reply]
@alan-nakagiri can you post your code? I think I have the same problem

**Reply by alan-nakagiri** [reply]
@DaviAraki I essentially had to import `react-hook-form` into my test to grab a `control` to pass into a mock component that was returning the component I was trying to test. 

```      
const MockForm = () => {
        const { control } = useForm()

        return (
          <Dropdown
            {...props}
            control={control}
          />
        )
      }

render(<MockForm />
```

**Reply by DaviAraki** [reply]
Worked, thanks
